evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -48,50 +48,60 @@ val matrix = javaClass.getResourceAsStream("/rules/osadl-matrix.json").use {
     Json.Default.decodeFromStream<Map<String, Map<String, String>>>(it)
 }.mapValues { (_, outbounds) -> outbounds.mapValues { (_, value) -> Compatibility.fromString(value) } }
 
-fun PackageRule.LicenseRule.isOsadlMatrixNonCompliant(inbound: SpdxSingleLicenseExpression) =
+fun PackageRule.LicenseRule.isOsadlMatrixNonCompliant(outbound: SpdxSingleLicenseExpression) =
     object : RuleMatcher {
         override val description = "isOsadlMatrixCompliant($license)"
 
         @ExperimentalSerializationApi
         override fun matches(): Boolean {
             // Be conservative and use the simple license string without the exception string for lookup. Eventually
             // resulting false-positives are better than an incompatibility to be missed.
-            val outbound = license.simpleLicense()
-            return matrix.getValue(inbound.simpleLicense()).getValue(outbound) == Compatibility.NO
+            val inbound = license.simpleLicense()
+            return matrix.getValue(inbound).getValue(outbound.simpleLicense()) == Compatibility.NO
         }
     }
 
 val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+
     // Define a rule that is executed for each dependency.
     dependencyRule("OSADL compliance matrix") {
         // Requirements for the rule to trigger a violation.
         require {
             -isExcluded()
         }
 
+        val projectLicenseInfo = licenseInfoResolver.resolveLicenseInfo(project.id).filter(licenseView)
```

Comment:
I don't understand where `project.id` / `project` is defined. Can you give a hint?

Reply:
It's part of the `DependencyRule` itself:

https://github.com/oss-review-toolkit/ort/blob/22640c2bfe89692bec1a23660ec006469d51f4a0/evaluator/src/main/kotlin/DependencyRule.kt#L63-L66

So a dependency rules iterates over all dependencies *of all projects*, and we can use `project` to determinie what the "current" project is.

Reply:
Thanks!